types/helper.go

```diff
@@ -23,31 +23,35 @@ import (
 	"github.com/pingcap/errors"
 )
 
-// RoundFloat rounds float val to the nearest even integer value with float64 format, like MySQL Round function.
-// RoundFloat uses default rounding mode, see https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html
-// so rounding use "round to nearest even".
-// e.g, 1.5 -> 2, -1.5 -> -2.
-func RoundFloat(f float64) float64 {
-	return math.RoundToEven(f)
-}
-
-// Round rounds the argument f to dec decimal places.
+// RoundFloat rounds the argument f to dec decimal places using "round to nearest even" rule.
 // dec defaults to 0 if not specified. dec can be negative
-// to cause dec digits left of the decimal point of the
-// value f to become zero.
-func Round(f float64, dec int) float64 {
+// see https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html
+func RoundFloat(f float64, dec int) float64 {
 	shift := math.Pow10(dec)
 	tmp := f * shift
 	if math.IsInf(tmp, 0) {
 		return f
 	}
-	result := RoundFloat(tmp) / shift
+	result := math.RoundToEven(tmp) / shift
 	if math.IsNaN(result) {
 		return 0
 	}
 	return result
 }
 
+// RoundInt rounds the argument i to dec decimal places using "round half up" rule.
+// dec defaults to 0 if not specified. dec can be negative
+func RoundInt(i int64, dec int) int64 {
+	// is itself when dec >= 0
+	if dec >= 0 {
+		return i
+	}
+
+	shift := math.Pow10(-dec)
+	intPart := math.Round(float64(i) / shift)
```

Comment:
Add a test case.
```
MySQL [test]>  select round(24999999999999999,-16);
+------------------------------+
| round(24999999999999999,-16) |
+------------------------------+
|            20000000000000000 |
+------------------------------+
1 row in set (0.000 sec)
```


We should avoid float calc in int round.

Reply:
added and failed, will fix later